plugins/modules/backup_plan_info.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+DOCUMENTATION = r"""
+---
+module: backup_plan_info
+version_added: 6.0.0
+short_description: Describe AWS Backup Plans
+description:
+  - Lists info about Backup Plan configuration.
+author:
+  - Gomathi Selvi Srinivasan (@GomathiselviS)
+  - Kristof Imre Szabo (@krisek)
+  - Alina Buzachis (@alinabuzachis)
+options:
+  backup_plan_names:
+    type: list
+    elements: str
+    default: []
+    description:
+      - Specifies a list of plan names.
+      - If an empty list is specified, information for the backup plans in the current region is returned.
```

Comment:
I guess we should add note like "Listing all the AWS backup plans is not supported at the moment. When I(backup_plan_names) is not set, an empty list is returned." We can then add this functionality next.

Reply:
@tremble What do you suggest?

Reply:
Keep it simple.  Make it required for now, we can make it optional later as a "new feature" (non-breaking IMO).

Reply:
Right, no breaking changes! Makes sense.

Reply:
My bad! I guess that functionality works https://github.com/ansible-collections/amazon.aws/pull/1446/files#diff-d2301559aaf1b18099212d421d58fc78b99063e3f730e0c23c1cc79cff553402R114-R121 (missed this part), @hakbailey can you please confirm?